eq-author/src/components/CommentsPanel/index.test.js

```diff
@@ -0,0 +1,221 @@
+        },
+      },
+    };
+
+    mocks = [
+      {
+        request: {
+          query: COMMENT_QUERY,
+          variables: {
+            input: { pageId: "P1" },
+          },
+        },
+        result: () => {
+          queryWasCalled = true;
+          return {
+            data: {
+              page: {
+                id: "P1",
+                comments: [
+                  {
+                    commentText: "Query comment body",
+                    createdTime: "2019-10-16T07:39:46.984Z",
+                    id: "C1",
+                    replies: [],
+                    user: {
+                      displayName: "Fred Jones",
+                      email: "test@tester.com",
+                      id: "U1",
+                      name: "Fred Jones",
+                      picture: null,
+                      __typename: "User",
+                    },
+                    __typename: "Comment",
+                  },
+                ],
+                __typename: "QuestionPage",
+              },
+            },
+          };
+        },
+      },
+
+      {
+        request: {
+          query: COMMENT_ADD,
+          variables: {
+            input: {
+              pageId: "P1",
+              commentText: "This is a test ADD comment",
+            },
+          },
+        },
+        result: () => {
+          queryWasCalled = true;
+          return {
+            data: {
+              createComment: {
+                id: "C1",
+                commentText: "This is a test ADD comment",
+                createdTime: "2019-10-17T07:15:19.229Z",
+                page: {
+                  id: "P1",
+                  comments: [
+                    {
+                      id: "C1",
+                      __typename: "Comment",
+                    },
+                  ],
+                  __typename: "QuestionPage",
+                },
+                user: {
+                  id: "U1",
+                  name: "Fred Jones",
+                  picture: null,
+                  email: "test@tester.com",
+                  displayName: "Fred Jones",
+                  __typename: "User",
+                },
+                replies: [],
+                __typename: "Comment",
+              },
+            },
+          };
+        },
+      },
+      {
+        request: {
+          query: COMMENT_QUERY,
+          variables: {
+            input: { pageId: "P2" },
+          },
+        },
+        result: () => {
+          queryWasCalled = true;
+          return {
+            data: {},
+            errors: ["Oops! Something went wrong"],
+          };
+        },
+      },
+    ];
+  });
+
+  /////////////////////////////////////////////////////////////////////////
+
+  it("should render comments txt area", async () => {
+    const { getByTestId } = render(<CommentsPanel {...props} />, {
+      route: "/q/Q1/page/P1",
+      urlParamMatcher: "/q/:questionnaireId/page/:pageId",
+      mocks,
+    });
```

Comment:
We render the same thing in every test, could we extract it?

Reply:
done